### DISCUSSION TITLE
Follow-up on Implementation of custom layer in Lux

### DISCUSSION THREAD

**Discussion by ghost** [LuxDL/Lux.jl#378] [Q&A]
Hi, 

This question is a follow up to the issue https://github.com/LuxDL/Lux.jl/issues/376.

A quick follow up on `mat = reshape(y, 3, 3)` syntax.
How would it handle a symmetric matrix if I supply 6 independent components?

I know from symmetry that 
`mat=[y[1] y[2] y[3]; y[2] y[4] y[5]; y[3] y[5] y[6]]`.  Would this be acceptable for AD? 

Additionally, how can I check is AD will compute the gradients? 

Best Regards

**Comment by avik-pal** [comment]
The default AD (Zygote) will not be able to differentiate that code. You will have to define a custom `rrule`. See https://juliadiff.org/ChainRulesCore.jl/dev/rule_author/which_functions_need_rules.html fro a good reference.
